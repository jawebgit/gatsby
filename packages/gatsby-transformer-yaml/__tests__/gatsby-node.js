const Promise = require("bluebird")
const yaml = require("js-yaml")

const { onNodeCreate } = require("../src/gatsby-node")

describe(`Process YAML nodes correctly`, () => {
  const node = {
    id: "whatever",
    contentDigest: "whatever",
    mediaType: `text/yaml`,
    children: [],
    name: `test`,
  }

  // Make some fake functions its expecting.
  const loadNodeContent = node => {
    return Promise.resolve(node.content)
  }

  it(`correctly creates nodes from JSON which is an array of objects`, async () => {
    const data = [{ blue: true, funny: "yup" }, { blue: false, funny: "nope" }]
    node.content = yaml.safeDump(data)

    const createNode = jest.fn()
    const updateNode = jest.fn()
    const boundActionCreators = { createNode, updateNode }

    await onNodeCreate({
      node,
      loadNodeContent,
      boundActionCreators,
    }).then(() => {
      expect(createNode.mock.calls).toMatchSnapshot()
      expect(updateNode.mock.calls).toMatchSnapshot()
      expect(createNode).toHaveBeenCalledTimes(2)
      expect(updateNode).toHaveBeenCalledTimes(1)
    })
  })

  it(`If the object has an id, it uses that as the id instead of the contentDigest`, async () => {
    const data = [
      { id: "foo", blue: true, funny: "yup" },
      { blue: false, funny: "nope" },
    ]
    node.content = yaml.safeDump(data)

    const createNode = jest.fn()
    const updateNode = jest.fn()
    const boundActionCreators = { createNode, updateNode }

    await onNodeCreate({
      node,
      loadNodeContent,
      boundActionCreators,
    }).then(() => {
      expect(createNode.mock.calls[0][0].id).toEqual("foo")
    })
  })
})
