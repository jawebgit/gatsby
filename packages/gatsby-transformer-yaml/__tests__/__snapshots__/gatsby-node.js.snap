// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Process YAML nodes correctly correctly creates nodes from JSON which is an array of objects 1`] = `
Array [
  Array [
    Object {
      "blue": true,
      "children": Array [],
      "content": "{\\"blue\\":true,\\"funny\\":\\"yup\\"}",
      "contentDigest": "73901821b17d5aa9dd6026181f73b64c",
      "funny": "yup",
      "id": "whatever [0] >>> YAML",
      "mediaType": "application/json",
      "parent": "whatever",
      "type": "Test",
    },
  ],
  Array [
    Object {
      "blue": false,
      "children": Array [],
      "content": "{\\"blue\\":false,\\"funny\\":\\"nope\\"}",
      "contentDigest": "f624311d932d73dcd416d2a8bea2b67d",
      "funny": "nope",
      "id": "whatever [1] >>> YAML",
      "mediaType": "application/json",
      "parent": "whatever",
      "type": "Test",
    },
  ],
]
`;

exports[`Process YAML nodes correctly correctly creates nodes from JSON which is an array of objects 2`] = `
Array [
  Array [
    Object {
      "children": Array [
        "whatever [0] >>> YAML",
        "whatever [1] >>> YAML",
      ],
      "content": "- blue: true
  funny: yup
- blue: false
  funny: nope
",
      "contentDigest": "whatever",
      "id": "whatever",
      "mediaType": "text/yaml",
      "name": "test",
    },
  ],
]
`;
